obfuscate.h

```diff
@@ -73,9 +80,14 @@ namespace ay
 
 	template <typename T>
 	using char_type = typename remove_const_ref<T>::type;
+#endif
 
 	// Generate a pseudo-random key that spans all 8 bytes
+	#if __cplusplus >= 202002L
+	consteval key_type generate_key(key_type seed)
+	#else
 	constexpr key_type generate_key(key_type seed)
+	#endif
```

Comment:
I'd avoid this kind of branching all over the place. Instead, define a macro like `_AY_CONST` or similar and then use this instead.

```cpp
#if __cplusplus >= 202002L
    #define _AY_CONST consteval
#else
    #define _AY_CONST constexpr
#endif

_AY_CONST key_type generate_key(key_type seed)
{
    // ...
}

_AY_CONST size_type size() const
{
    // ...
}
```

Reply:
Thanks for doing this!

I agree, but would probably give it a more explicit name like AY_CONSTEVAL.

Reply:
Sure, that will make the code a cleaner. Will update it.